criu/cr-restore.c

```diff
@@ -3128,6 +3128,24 @@ static void rst_reloc_creds(struct thread_restore_args *thread_args,
 	thread_args->creds_args = args;
 }
 
+static bool groups_are_okay(gid_t* groups, int n_groups) {
+	int n, len;
+	bool ret;
+	gid_t* gids;
+
+	n = getgroups(0, NULL);
+	if (n != n_groups)
+		return false;
+	if (n == 0)
+		return true;
+	len = n * sizeof(gid_t);
+	gids = (gid_t*)xmalloc(len);
```

Comment:
There is no need to cast `xmalloc` return value.
But, it should be checked for `!NULL`.

Reply:
Good catch. Thank you!